test/client/spec/victory-util/collection.spec.js

```diff
@@ -174,4 +177,71 @@ describe("collections", () => {
       expect(Collection.allSetsEqual(comparisons)).to.eql(false);
     });
   });
+
+
+  describe("areVictoryPropsEqual", () => {
+    let sandbox;
+    beforeEach(() => {
+      sandbox = sinon.sandbox.create();
+      sandbox.spy(Collection, "checkEquality");
+    });
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it("returns early when nested collections are strictly equal", () => {
+      const a = { test: { nested: "a" } };
+      const b = a;
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledOnce.and.returned(true);
+    });
+
+    it("returns early when nested collections are not the same length", () => {
+      const a = { test: { nested: "a" }, test2: { nested: "b" } };
+      const b = { test: { nested: "a" } };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(false);
+      expect(Collection.checkEquality).calledOnce.and.returned(false);
+    });
+
+    it("returns early if values are not the same type", () => {
+      const a = { test: { nested: "a" } };
+      const b = { test: () => "a" };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(false);
+      expect(Collection.checkEquality).calledOnce.and.returned(false);
+    });
+
+    it("returns early when nested elements are empty", () => {
+      const a = { a: [] };
+      const b = { a: [] };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledOnce.and.returned(true);
+    });
+
+    it("returns true if values are functions", () => {
+      const a = { test: () => "a" };
+      const b = { test: () => "a" };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+    });
+
+    it("recursively checks equality for nested collections", () => {
+      const a = { test: { nested: "a" }, test2: { nested: "a" } };
+      const b = { test: { nested: "a" }, test2: { nested: "a" } };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledThrice;
+    });
+
+    it("does not recursively check date objects", () => {
```

Comment:
This doesn't seem to be the right test title. These dates _are_ being compared, and there isn't a nested object. As luck has it, they're all equal, but aren't necessarily.

We probably need to (1) retitle and/or refactor this test, and (2) make sure to use fake timers in our sandbox, amending the`sandbox.create()` call above to:

```js
      sandbox = sinon.sandbox.create({
        useFakeTimers: true
      });
```